docs/reference/settings/security-settings.asciidoc

```diff
@@ -82,6 +83,33 @@ resource. When set to `false`, an HTTP 401 response is returned and the user
 can provide credentials with the appropriate permissions to gain
 access. Defaults to `true`.
 
+[float]
+[[security-automata-settings]]
+==== Automata Settings
+In places where {security} accepts wildcard patterns (e.g. index patterns in
+roles, group matches in the role mapping API), each pattern is compiled into
+an Automaton. The follow settings are available to control this behaviour.
+
+`xpack.security.automata.max_determinized_states`::
+The upper limit on how many automaton states may be created by a single pattern.
+This protects against too-difficult (e.g. exponentially hard) patterns.
+Defaults to `100,000`.
+
+`xpack.security.automata.cache.enabled`::
+Whether to cache the compiled automata. Compiling automata can be CPU intensive
+and may slowdown some operations. The cache reduces the frequency with which
+automata need to be compiled.
+Defaults to `true`.
+
+`xpack.security.automata.cache.size`::
+The maximum number of items to retain in the automata cache.
+Defaults to `500`.
```

Comment:
500 seems low to me given the places we use automata. Any number we pick will be somewhat arbitrary but my gut says 10,000. Also, if we are concerned about excess memory usage, the lucene automata implement Accountable which provides `ramBytesUsed` so we can also limit by amount of memory/percent of heap in the cache.